tests/_internal/pydantic/test_model_validator.py

```diff
@@ -0,0 +1,296 @@
+import pytest
+from pydantic import BaseModel, ValidationError
+
+from prefect._internal.pydantic import HAS_PYDANTIC_V2
+from prefect._internal.pydantic._flags import USE_V2_MODELS
+from prefect._internal.pydantic.utilities.model_validator import model_validator
+
+
+@pytest.mark.skipif(
+    HAS_PYDANTIC_V2,
```

Comment:
should this be `USE_V2_MODELS`?

Reply:
hm, so it seems like there's an issue when it `HAS_PYDANTIC_V2` but doesn't have the env var enabled. It doesn't hit the validators at all 🤔

Reply:
as it turns out, I needed to import the `prefect.pydantic` `BaseModel` to use the "v2v1" model for the case where `HAS_PYDANTIC_V2` and not `USE_PYDANTIC_V2` because the flags don't affect that. applied in [030aa2e](https://github.com/PrefectHQ/prefect/pull/12635/commits/030aa2e9b04fd7baac6c443fa8d89b00f7670f8e)

Reply:
can you explain why this is true?
> import the prefect.pydantic BaseModel to use the "v2v1" model for the case where HAS_PYDANTIC_V2 and not USE_PYDANTIC_V2 because the flags don't affect that

this is counter-intuituve for me looking at `_compat.py`

Reply:
I would agree with that, it is counter-intuitive to me as well